File: app/src/main/java/com/lucasurbas/counter/ui/explore/usecase/GetCountersUpdatesInteractor.java
package com.lucasurbas.counter.ui.explore.usecase;

import com.lucasurbas.counter.app.di.rx.InteractorSchedulers;
import com.lucasurbas.counter.data.model.Counter;
import com.lucasurbas.counter.data.repository.CounterRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

public class GetCountersUpdatesInteractor {

    private final CounterRepository counterRepository;
    private final InteractorSchedulers interactorSchedulers;

    @Inject
    public GetCountersUpdatesInteractor(CounterRepository counterRepository,
                                        InteractorSchedulers interactorSchedulers){
        this.counterRepository = counterRepository;
        this.interactorSchedulers = interactorSchedulers;
    }

    public Observable<List<Counter>> getCountersUpdates(){
        return counterRepository.getCountersUpdates()
                .subscribeOn(interactorSchedulers.getBackgroundScheduler())
                .observeOn(interactorSchedulers.getMainThreadScheduler());
    }
}

File: app/src/main/java/com/lucasurbas/counter/ui/explore/usecase/GetCountersUpdatesInteractor.kt
package com.lucasurbas.counter.ui.explore.usecase

import com.lucasurbas.counter.app.di.rx.InteractorSchedulers
import com.lucasurbas.counter.data.model.Counter
import com.lucasurbas.counter.data.repository.CounterRepository
import io.reactivex.Observable
import javax.inject.Inject

class GetCountersUpdatesInteractor @Inject constructor(
        private val counterRepository: CounterRepository,
        private val interactorSchedulers: InteractorSchedulers) {

    fun getCountersUpdates(): Observable<List<Counter>> {
        return counterRepository.countersUpdates
                .subscribeOn(interactorSchedulers.backgroundScheduler)
                .observeOn(interactorSchedulers.mainThreadScheduler)
    }
}
